test/jit/test_pdt.py

```diff
@@ -224,3 +224,68 @@ def test_model_with_args(a, m):
         user_class = class_with_args(False)
         scripted_fn = torch.jit._script_pdt(test_model_with_args, example_inputs=[(10, user_class, ), (10.9, user_class, ), ])
         self.assertEqual(scripted_fn(100, class_with_args(True), ), test_model_with_args(100, class_with_args(True)))
+
+    def test_nn_module(self):
+        class TestPDTModel(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+
+            def forward(self, x) -> Any:
+                if isinstance(x, int):
+                    return x + 1
+                elif isinstance(x, float):
+                    return x - 1
+                else:
+                    return x
+
+        make_global(TestPDTModel)
+        pdt_model = TestPDTModel()
+        scripted_pdt_model = torch.jit._script_pdt(pdt_model, example_inputs=[(10, ), (10.80, ), (False, )])
+        self.assertEqual(scripted_pdt_model(50), pdt_model(50))
+        self.assertEqual(scripted_pdt_model(1.8), pdt_model(1.8))
+        self.assertTrue(scripted_pdt_model(True), pdt_model(True))
+
+    def test_nested_nn_module_class(self):
+        class NestedPDTInner(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+
+            def forward(self, x):
+                if isinstance(x, int):
+                    return x * 10
+                return x
+
+        class NestedModulePDTWrapper(torch.nn.Module):
+            def __init__(self, inner):
+                super().__init__()
+                self.inner = inner
+
+            def forward(self, x):
+                return self.inner(x)
+
+        make_global(NestedPDTInner, NestedModulePDTWrapper)
+        inner_pdt_model = NestedPDTInner()
+        wrapped_pdt_model = NestedModulePDTWrapper(inner_pdt_model)
+        scripted_pdt_model = torch.jit._script_pdt(wrapped_pdt_model, example_inputs=[(20, ), (False, )])
+        self.assertEqual(scripted_pdt_model(30), wrapped_pdt_model(30))
+        self.assertEqual(scripted_pdt_model(1.9), wrapped_pdt_model(1.9))
+        self.assertTrue(scripted_pdt_model(True), wrapped_pdt_model(True))
+
+    def test_nested_function_in_forward(self):
+        class NestedFunctionInForward(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+
+            def forward(self, x):
+                return self.fun(x) + 10
+
+            def fun(self, x):
+                if isinstance(x, int):
+                    return x + 1
+                return 0
+
+        make_global(NestedFunctionInForward)
+        pdt_model = NestedFunctionInForward()
+        scripted_pdt_model = torch.jit._script_pdt(pdt_model, example_inputs=[(20, ), (False, )])
+        self.assertEqual(scripted_pdt_model(30), pdt_model(30))
+        self.assertEqual(scripted_pdt_model(10.9), pdt_model(10.9))
```

Comment:
There are no example inputs of `float` type, why is it an allowed input type?
And could you add an assert about `boolean` type inputs, which is part of example input set?

Reply:
There are no example inputs of float type, why is it an allowed input type?
            >> Because, the type gets annotated to `Any` and thus `float` becomes a default `float` type. This will be the case until `Any` is changed to `Union`. I updated the test case to now take `int` and `bool` for now.